makePoplogCommander.sh

```diff
@@ -25,6 +27,92 @@ cat << \****
 #define PREFER_FLEXIBILITY  (RUN_INIT_P|INHERIT_ENV)
 
 
+///////////////////////////////////////////////////////////////////////////////
+//  Library routines: 
+//      Mishap, an error reporting function
+//      Chain, a 1D vector
+///////////////////////////////////////////////////////////////////////////////
+
+
+//  Mishap - Error reporting using printf-like functionality.
+
+void mishap( const char *msg, ... ) {
+    va_list args;
+    va_start( args, msg );
+    fprintf( stderr, "Mishap: " );
+    vfprintf( stderr, msg, args );
+    fprintf( stderr, "\n" );
+    va_end( args );
+    exit( EXIT_FAILURE );
+}
+
+//  Chains - 1D vectors.
+
+typedef struct Chain *Chain;
+typedef char * Ref;
```

Comment:
Why `char *` and not `void *`? Is `Chain` not meant to be generic in the elements it holds?

Reply:
There's no good reason - the code was copied from elsewhere and adapted. I did think about this but didn't get around to making the change. I will fix it.

Reply:
Fixed.